engine/runtime/src/main/scala/org/enso/compiler/pass/analyse/ImportSymbolAnalysis.scala

```diff
@@ -0,0 +1,128 @@
+package org.enso.compiler.pass.analyse
+
+import org.enso.compiler.context.{InlineContext, ModuleContext}
+import org.enso.compiler.core.IR
+import org.enso.compiler.data.BindingsMap
+import org.enso.compiler.pass.IRPass
+import org.enso.compiler.pass.desugar.GenerateMethodBodies
+
+import scala.annotation.unused
+
+/** Performs analysis of `from ... import sym1, sym2, ...` statements - checks that all
+  * the symbols imported from the module can be resolved, i.e., exists.
+  * In case of unresolved symbols, replaces the IR import with [[IR.Error.ImportExport]].
+  * Reports only the first unresolved symbol.
+  */
+case object ImportSymbolAnalysis extends IRPass {
+
+  override type Metadata = BindingsMap
+
+  override type Config = IRPass.Configuration.Default
+
+  override val precursorPasses: Seq[IRPass] =
+    Seq(BindingAnalysis, GenerateMethodBodies)
+
+  override val invalidatedPasses: Seq[IRPass] =
+    Seq()
+
+  /** @inheritdoc
+    */
+  override def runModule(
+    ir: IR.Module,
+    moduleContext: ModuleContext
+  ): IR.Module = {
+    val bindingMap = ir.unsafeGetMetadata(
+      BindingAnalysis,
+      "BindingMap should already be present"
+    )
+    ir.copy(
+      imports = ir.imports.flatMap(analyseSymbolsFromImport(_, bindingMap))
+    )
+  }
+
+  /** @inheritdoc
+    */
+  override def runExpression(
+    ir: IR.Expression,
+    inlineContext: InlineContext
+  ): IR.Expression = ir
+
+  /** @return May return multiple [[IR.Error.ImportExport]] in case of multiple unresolved symbols.
+    */
+  private def analyseSymbolsFromImport(
+    imp: IR.Module.Scope.Import,
+    bindingMap: BindingsMap
+  ): List[IR.Module.Scope.Import] = {
+    imp match {
+      case imp @ IR.Module.Scope.Import.Module(
+            _,
+            _,
+            _,
+            Some(onlyNames),
+            _,
+            _,
+            _,
+            _,
+            _
+          ) =>
+        bindingMap.resolvedImports.find(_.importDef == imp) match {
+          case Some(resolvedImport) =>
+            val importedTarget = resolvedImport.target
+            val unresolvedSymbols =
+              onlyNames.filterNot(isSymbolResolved(importedTarget, _))
+            if (unresolvedSymbols.nonEmpty) {
+              unresolvedSymbols
+                .map(
+                  createErrorForUnresolvedSymbol(
+                    imp,
+                    importedTarget,
+                    _
+                  )
+                )
+            } else {
+              List(imp)
+            }
+          case None => List(imp)
```

Comment:
So if we don't find the import we return.. the import? Shouldn't that return an error?

Reply:
I am not sure, to be honest. Usually, `bindingMap.resolvedImports.find(_.importDef == imp)` finds something. The only case when this will return `None`, is when `ImportResolver` was not run previously, which might happen, e.g., in a testing environment. What do you think? Should that return an error?

Previously, I had there `bindingMap.resolvedImports.find(_.importDef == imp).getOrElse(throw new CompilerError(...)`, but that failed in the testing environment.